core/project_settings.h

```diff
@@ -201,6 +209,13 @@ class ProjectSettings : public Object {
 	// There is therefore the potential for a change to be missed. Persisting the counter
 	// for two frames avoids this, at the cost of a frame delay.
 	bool has_changes() const { return _dirty_this_frame == 1; }
+
+	// Testing a version allows fast cached GET_GLOBAL macros.
+	uint32_t get_version() const { return _version; }
+
+	void _get_global_lock() { _get_global_mutex.lock(); }
+	void _get_global_unlock() { _get_global_mutex.unlock(); }
```

Comment:
I think i'd rather expose `_get_global_mutex` with `Mutex& get_global_mutex`, for callers to be able to use `MutexLock` for RAII safety.